client/src/api/navMeshSlice.ts

```diff
@@ -22,12 +22,15 @@ const navMeshSlice = createSlice({
   name: "navmesh",
   initialState: initialState,
   reducers: {
-    updated: (state, action: PayloadAction<INavMeshUpdate>) => {
-      const polys = action.payload.mesh.polys;
-      if (action.payload.blue) {
-        state.blue = polys;
-      } else {
-        state.red = polys;
+    updated: (state, action: PayloadAction<INavMeshUpdate[]>) => {
+      for (const [blue, navmesh] of Object.entries(action.payload)) {
+        const data = {blue: (blue === "true"), mesh: navmesh} as unknown as INavMeshUpdate
```

Comment:
What's going on here?

Reply:
TSC being very annoying...

Python has no problems making a dictionary that maps a boolean to some object, but typescript doesn't want to play ball. It replaces it by taking the boolean value and transforming it to a string that maps to the navmesh. When I try recasting without the "as unknown" part, TSC complains that navmesh has no property "polys", though I know for a fact it does. It's part of the ugliness I described on discord...

Reply:
Can you use a record instead? Object keys must be numbers or strings, but the `Record` type is a true key/value pair.

Reply:
> Can you use a record instead? Object keys must be numbers or strings, but the `Record` type is a true key/value pair.

It seems like that doesn't change much, TSC is still asking for a number, string or symbol for the keys. Suppose I leave the "as unknown" in the NavMeshes part then I get the following error:

> TS2352: Conversion of type '{ blue: boolean; mesh: INavMeshUpdate; }' to type 'INavMeshUpdate' may be a mistake because neither type sufficiently overlaps with the other. <ins>If this was intentional, convert the expression to 'unknown' first.</ins>
Types of property 'mesh' are incompatible.
Property 'polys' is missing in type 'INavMeshUpdate' but required in type 'NavMesh'.

So TSC is literally suggesting the cast-to-unknown solution, though I'm definitely not a fan of this either...